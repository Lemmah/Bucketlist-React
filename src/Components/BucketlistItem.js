import React, { Component } from 'react'


class BucketlistItem extends Component{

	displayBucketlistItem(){
		if(!this.props.bucketlistItems){
			return(
				<ul className="list-group">
				  <li className="list-group-item"></li>
				  <li className="list-group-item">
				  	This Bucketlist is empty.
				  </li>
				  <li className="list-group-item"></li>
				</ul>
			);
		} else {
			return (
				<ul className="list-group"> 
						<h3>
							<center>
								{this.props.bucketlistName}
							</center>
						</h3>
					{this.props.bucketlistItems.map((bucketlistItem) => {
					   return (
					   	<li
					   			key={bucketlistItem.id}
					   			className="list-group-item bucketlistItem"
					   	>{bucketlistItem.name}</li>);
					})}
				</ul>
			);
		}
	}

	render(){
		return(
			<div>
				{this.displayBucketlistItem()}
			</div>
		);
	}
}

export default BucketlistItem;