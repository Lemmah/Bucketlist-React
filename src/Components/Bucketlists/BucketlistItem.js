import React, { Component } from 'react'
import { Button, ButtonToolbar, Row, Col } from 'react-bootstrap'

const buttonsInstance = (
  <ButtonToolbar>
    <Button>A</Button>
    <Button bsStyle="primary">B</Button>
    <Button bsStyle="success">C</Button>
    <Button bsStyle="info">D</Button>
    <Button bsStyle="warning">E</Button>
    <Button bsStyle="danger">F</Button>
  </ButtonToolbar>
);

class BucketlistItem extends Component{

	displayBucketlistItem(){
		if(!this.props.bucketlistName){
			return (
				<ul className="list-group">
					<h3>
							<center>
								Welcome! Before you die...?
							</center>
						</h3>
				  <li className="list-group-item"></li>
				  <li className="list-group-item">
				  	You currently don't have any bucketlists! Maybe you should
				  	start by creating one then you items in it will appear here.
				  </li>
				  <li className="list-group-item"></li>
				</ul>
			);
		} else {
			if(!this.props.bucketlistItems.length > 0){
				return(
					<ul className="list-group">
						<h3>
							<center>
								{this.props.bucketlistName}
							</center>
						</h3>
					  <li className="list-group-item"></li>
					  <li className="list-group-item">
					  	There are no activities in here.
					  </li>
					  <li className="list-group-item"></li>
					</ul>
				);
			} else {
				return (
					<ul className="list-group">
						<Row>
			      	<Col xs={6} md={6}>
			      		<h3>
									<center>
										{this.props.bucketlistName}
									</center>
								</h3>
			      	</Col>
			      	<Col xs={6} md={6}>
			      		{buttonsInstance}
			      	</Col>
		      	</Row> 
						{this.props.bucketlistItems.map((bucketlistItem) => {
						   return (
						   	<li
						   			key={bucketlistItem.id}
						   			className="list-group-item bucketlistItem"
						   	>
						   	<Row>
						   		<Col md={6} sm={6}>
						   		{bucketlistItem.name}
						   		</Col>
						   		<Col md={6} sm={6} style={{textAlignment: "right"}}>{' '}
						   		<Button bsStyle="success">A</Button>{' '}
						   		<Button bsStyle="warning">E</Button>{' '}
						   		<Button bsStyle="danger">D</Button>{' '}
						   		</Col>
						   	</Row>
						   	</li>);
						})}
					</ul>
				);
			}
		}
	}

	render(){
		return(
			<div>
				{this.displayBucketlistItem()}
			</div>
		);
	}
}

export default BucketlistItem;