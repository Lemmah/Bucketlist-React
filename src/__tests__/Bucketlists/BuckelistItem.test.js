import React from 'react';
import renderer from 'react-test-renderer';
import BucketlistItem from '../../Components/Bucketlists/BucketlistItem';

it('Displays BucketlistItem without Crashing', () => {
	const rendered = renderer.create(
		<BucketlistItem />,
		);
	expect(rendered.toJSON()).toMatchSnapshot();
})