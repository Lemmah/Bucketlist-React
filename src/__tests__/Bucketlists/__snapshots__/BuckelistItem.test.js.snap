// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays BucketlistItem without Crashing 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          <h3>
             Welcome, before you die? 
          </h3>
        </th>
        <th>
          Create Some Bucketlists
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <td>
          Yo! Start making your life meaningful!
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <br />
        </td>
        <td>
          <br />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
